sdk/assertion.go

```diff
@@ -1,4 +1,313 @@
 package sdk
 
-type Assertion struct {
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// AssertionType - Custom type to hold value for the assertion ranging from 0-1
+type AssertionType uint
+
+const (
+	handlingAssertion AssertionType = iota + 1
+	baseAssertion
+)
+
+var (
+	AssertionTypeName = map[uint8]string{
+		1: "Handling",
+		2: "Base",
+	}
+	AssertionTypeValue = map[string]uint8{
+		"Handling": 1,
+		"Base":     2,
+	}
+)
+
+func (a AssertionType) String() string {
+	return AssertionTypeName[uint8(a)]
+}
+
+func ParseAssertionType(a string) (AssertionType, error) {
+	a = strings.TrimSpace(a)
+	value, ok := AssertionTypeValue[a]
+	if !ok {
+		return AssertionType(0), fmt.Errorf("%q is not a valid assertion type", a)
+	}
+	return AssertionType(value), nil
+}
+
+func (a AssertionType) MarshalJSON() ([]byte, error) {
+	return json.Marshal(a.String())
+}
+
+func (a *AssertionType) UnmarshalJSON(data []byte) error {
+	var assertionType string
+	err := json.Unmarshal(data, &assertionType)
+	if err != nil {
+		return err
+	}
+	if *a, err = ParseAssertionType(assertionType); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Scope - Custom type to hold value for the assertion ranging from 0-1
+type Scope uint
+
+const (
+	trustedDataObj Scope = iota + 1
+	paylaod
+	explicit
+)
+
+var (
+	ScopeName = map[uint8]string{
+		1: "TDO",
+		2: "PAYL",
+		3: "EXPLICIT",
```

Comment:
@ttschampel @biscoe916 We want to remove explicit right?
```suggestion
```

Reply:
Yes, I don't think we need explicit.